### PR TITLE
[WIP] Move LoadPodFromFile to //pkg/apis/core/v1/helper

### PR DESCRIPTION
**Pull request by rosti**
**What this PR does / why we need it**:
LoadPodFromFile currently resides in //pkg/volume/util. However, it is not
volume specific function and some third parties (most notably kubeadm) are
importing the whole of //pkg/volume/util just to use it. This causes a lot
of unneeded dependencies to be included and the produced executable to bloated.
The solution is to move LoadPodFromFile to a more appropriate location and
//pkg/apis/core/v1/helper is the most obvious one.

For example, in the case of kubeadm, this change alone reduces imported packages from 623 to 551.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Refs kubernetes/kubeadm#839

**Special notes for your reviewer**:
/cc @kubernetes/sig-cluster-lifecycle-pr-reviews
/cc @kubernetes/sig-api-machinery-pr-reviews
/cc @kubernetes/sig-storage-pr-reviews

/assign @thockin
/assign @luxas
/assign @lavalamp

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by luxas — APPROVED** [review]
/lgtm
/approve
for my part at least. Thanks @rosti, this is a clean and efficient code move :clap:

**Comment by rosti** [comment]
Redirected to `//pkg/apis/core/v1/helper`

/cc @kubernetes/sig-apps-pr-reviews

**Comment by liggitt** [comment]
/hold
actually, we want to keep legacyscheme use from bleeding into the apis/core package. can you use the generated v1 scheme instead (it should have the types registered properly already)

File: pkg/apis/core/v1/helper/helpers.go [hunk 1]
```diff
@@ -486,3 +489,24 @@ func GetPersistentVolumeClaimClass(claim *v1.PersistentVolumeClaim) string {
...
+	podDef, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read file path %s: %+v", filePath, err)
+	}
+	if len(podDef) == 0 {
+		return nil, fmt.Errorf("file was empty: %s", filePath)
+	}
+	pod := &v1.Pod{}
+
+	codec := legacyscheme.Codecs.UniversalDecoder()
```
Comment (sttts) [inline pkg/apis/core/v1/helper/helpers.go]: this dependency is not possible in `pkg/apis` unfortunately

**Comment by sttts** [comment]
For the scheme you can also just use client-go's scheme in k8s.io/client-go/kubernetes/scheme. k8s.io/client-go/kubernetes/tools/manifests would be a good location for such a generic helper. 

As a possible, worthwhile follow-up: we have plenty of manifest loaders in the code, follow the callers of https://github.com/sttts/kubernetes/blob/6a4d5cd7cc58e28c20ca133dab7b0e9e56192fe3/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/codec_factory.go#L184. All (or most) could use the same helper.

**Comment by rosti** [comment]
Thanks for the comments! I'll probably go the "generic file loader" func + using the scheme from client-go for this one.
Marking as a WIP until done.

**Comment by rosti** [comment]
I decided to replace the usage of `LoadPodFromFile` locally in kubeadm with a similar one (based on `LoadPodFromFile` itself). The PR (#64396) got merged, thus I close this.

**Pull request closed without merging**